### DISCUSSION TITLE
Exit code on syntax error?

### DISCUSSION THREAD

**Discussion by afaulconbridge** [awslabs/linter-rules-for-nextflow#32] [General]
Hello! I'm trying to use this via GitHub Actions with the Docker image to validate Nextflow scripts we'll be running on AWS Healthomics. 

I've noticed that when there is an invalid file, the exit code is still zero which indicates success. Is this something that can be addressed, or is this a CodeNarc issue? It seems as though there is a command line option `-failOnError=true` in [CodeNarc](https://codenarc.org/codenarc-command-line.html) for it

**Comment by markjschreiber** [comment]
CodeNarc is the application running the checks so it will be the one exiting with exit code `0`. Most linters try to recover from errors so they can lint as much of the file as they can understand. 

Technically speaking CodeNarc is run via this definition in the Docker container:

```
ENV ruleset "healthomics"
ENTRYPOINT bash check.sh ${ruleset}
```

The `check.sh` script is https://github.com/awslabs/linter-rules-for-nextflow/blob/main/scripts/check.sh

You could change to include the `--failOnError` and then rebuild the container from the `Dockerfile`. If that all works then feel free to submit a PR to update this. I can then also update the image on ECR public.

My only reservation would be that if the `--failOnError` immediately exits then that isn't great for general linting use, although it is useful for your use case.
